qiita_pet/uimodules/raw_data_tab.py

```diff
@@ -212,14 +175,26 @@ def render(self, prep, study_id, is_editable, ena_terms,
         preprocessing_status = prep.preprocessing_status
 
         if raw_data.filetype in ('SFF', 'FASTA'):
-            preprocess_form = Preprocess454ParametersForm()
+            all_params = Preprocessed454Params.get_all_parameters()
         elif raw_data.filetype == 'FASTQ':
-            preprocess_form = PreprocessIlluminaParametersForm()
+            all_params = PreprocessIlluminaParametersForm.get_all_parameters()
         else:
             raise ValueError("Don't know what to do but this exception will "
                              "never actually get shown anywhere because why "
                              "would you want to see tracebacks?")
 
+        preprocess_options = []
+        for param in all_params:
+            text = []
+            for k, v in viewitems(param):
+                if k in ('preprocessed_params_id', 'param_set_name'):
```

Comment:
Minor in this case since it is only two elements and the performance won't be terribly affected, but this tuple should be a set.